src/vocgui/list_filter.py

```diff
@@ -30,24 +30,28 @@ def lookups(self, request, model_admin):
         """
         list_of_disciplines = []
 
+        # Verify that only disciplines are displayed that actually can contain training sets
+        queryset = Discipline.objects.filter(
+            id__in=[
+                obj.id
+                for obj in Discipline.objects.all()
+                if obj.get_descendant_count() == 0
+            ]
+        )
```

Comment:
```suggestion
        # Verify that only disciplines are displayed that no not have child disciplines
        queryset = Discipline.objects.filter(lft=F("rght") - 1)
```
I'm not sure what you're trying to achieve here - if you really want disciplines with no child disciplines, you could improve performance by using [F()-](https://docs.djangoproject.com/en/4.0/topics/db/queries/#using-f-expressions-in-filters)objects and only query the disciplines from the database that you really want to have, instead of doing the additional database query to get all disciplines where you calculate whether they have children in Python.
The above suggestion works because in MPTT-models, the `lft` and `rght` fields are directly subsequent if the node does not have child nodes.